app/src/main/java/com/dongyang/android/youdongknowme/ui/view/search/SearchFragment.kt

```diff
@@ -15,6 +21,67 @@ class SearchFragment : BaseFragment<FragmentSearchBinding, SearchViewModel>() {
     }
 
     override fun initDataBinding() {
+        setupUI()
+        setTextClearButtonClickListener()
+    }
+
+    private fun setupUI() {
+        showKeyboardOnEditTextFocus()
+        setupHideKeyboardOnOutsideTouch()
+        setTextClearButtonVisibility()
+    }
+
+    private fun showKeyboardOnEditTextFocus() {
+        val searchKeyboard = binding.etSearchBar
+        searchKeyboard.requestFocus()
+
+        binding.etSearchBar.post {
+            binding.etSearchBar.setSelection(binding.etSearchBar.text.length)
+        }
```

Comment:
```suggestion
        binding.etSearchBar.requestFocus()
```

포커스만 줘도 키보드가 올라오지 않나요?

Reply:
```
binding.etSearchBar.post {
            binding.etSearchBar.setSelection(binding.etSearchBar.text.length)
        }
```

이 코드를 넣은 이유는 자동으로 텍스트의 길이에 맞춰 커서 위치를 끝으로 이동시키는 역할을 합니다. 

```
val searchKeyboard = binding.etSarchBar
```
는 제거해야겠군요 ㅎ,ㅎ

Reply:
해당 함수는 `setupUI()` 에서 호출되고 있는데요, 최초 진입시엔 입력한 텍스트가 없지 않을까요? 😅

Reply:
아하.............!